lib/beetle/dead_letter_queue.rb

```diff
@@ -0,0 +1,90 @@
+require 'net/http'
+require 'json'
+
+module Beetle
+  class DeadLetterQueue
+    READ_TIMEOUT = 3 #seconds
+    DEFAULT_DEAD_LETTER_MSG_TTL = 1000 #1 second
+    RABBIT_API_PORT = 15672
+
+    class FailedRabbitRequest < StandardError; end
+
+    class << self
+      def bind_dead_letter_queues!(channel, servers, target_queue, creation_keys = {})
```

Comment:
As discussed: Let's review the incoming creation_keys, if we have to handle/ignore e.g. per message ttl.